engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSharedApplication.h

```diff
@@ -0,0 +1,29 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_IOS_FRAMEWORK_SOURCE_FLUTTERSHAREDAPPLICATION_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_IOS_FRAMEWORK_SOURCE_FLUTTERSHAREDAPPLICATION_H_
+
+#import <UIKit/UIKit.h>
+
+@interface FlutterSharedApplication : NSObject
+
+/**
+ * Check whether the main bundle is an iOS App Extension.
+ */
++ (BOOL)isAppExtension;
```

Comment:
This should be a readonly class property.
```suggestion
@property(class, nonatomic, readonly) BOOL isAppExtension;
```